ddht/v5/crawl.py

```diff
@@ -0,0 +1,234 @@
+import contextlib
+import logging
+from typing import Iterator, List, Set
+
+from eth_enr import ENRAPI, ENRDB, ENRManager, default_identity_scheme_registry
+from eth_enr.exceptions import OldSequenceNumber
+from eth_typing import NodeID
+from eth_utils import encode_hex
+import trio
+
+from ddht.app import BaseApplication
+from ddht.boot_info import BootInfo
+from ddht.constants import DEFAULT_LISTEN, IP_V4_ADDRESS_ENR_KEY
+from ddht.exceptions import UnexpectedMessage
+from ddht.v5.app import get_local_private_key
+from ddht.v5.client import Client
+from ddht.v5.messages import FindNodeMessage
+
+logger = logging.getLogger("ddht.crawler")
+
+
+class StopScanning(Exception):
```

Comment:
Typically for this type of *internal* exception that is used in a closed-loop manner (it is raised and re-caught and should never make it out to userland), I've been naming them as `_StopScanning` to signal the internal nature.